docs/specs/#1984 - Apps and Packages Spec/spec.md

```diff
@@ -0,0 +1,118 @@
+
+### 1.3 Rough-in Designs
+
+![Apps and Packages View 1](apps-and-packages.png)
+
+![Update and View](update-and-view.png)
+
+## 2. Goals & User Cans
+
+### 2.1 Goals
+
+| No. | Goal | Priority |
+| --- | ---- | -------- |
+| 1   |   Provide users with an easy and delightful experience to manage installed applications.    |     0     |
+| 2   |   Incorporate advanced package management features such as version management, batch operations, automatic updates, and rollback capabilities.    |     0     |
+
+### 2.2 Non-Goals
+
+| No. | Non-Goal |
+| --- | -------- |
+| 1   |     This feature is not intended to replace the Windows Settings 'Apps > Installed' apps functionality.     |
+
+### 2.3 User Cans Summary Table
+
+| No. | User Can | Priority |
+| --- | -------- | -------- |
+| 1   |     A user can view application name, publisher, size, version, and source      |     0     |
+| 2   |    A user can switch between different versions of a package       |    0      |
+| 3   |    A user can select multiple applications at a time to perform bulk actions like update or uninstall       |    0      |
+| 4   |   A user can search to find an application       |      0    |
+| 5   |     A user can filter the page by update available      |      0    |
+| 6   |    A user can sort the page by application name, publisher, size, and source.      |    0      |
+
+## 3. User Stories
+
+### 3.1 User story - Updating and Rolling Back Packages
+
+A software developer wants to update a package to the latest version but finds that it has a bug affecting their system.
+
+#### Job-to-be-done
+
+1. Update a package to the latest version
+2. Roll back to a previous version of a package
+
+#### User experience
+
+- A software developer launches Dev Home and navigates to the Apps & Packages page.
+- They select the package that they want to update and click the Update button.
+- They are prompted to confirm the update and clicks Yes. The package is updated to the latest version.
+- They later find that the update has a security bug, so they go back to Dev Home to quickly revert to a previous version.
+- They select the package they want to roll back and click on the version.
+- They can see the available versions and select the version they want to roll back to.
+- The package is rolled back to the previous version.
+
+### 3.2 User story - Performing Bulk Actions
+
+A software developer wants to clean up space on their machine and uninstall large packages they are no longer using.
+
+#### Job-to-be-done
+
+1. Sort packages by size  
```

Comment:
What does `Size` here represent?  When most people look to free up space on their drive, people think  `Uninstalling X will free up Y GB`

Sounds like `Size` in this context is `The size of the app in its installed location` this could be smaller compared to the amount of space reclaimed by removing the package.

Reply:
I vote it should be the same size shown in Settings > Apps > Installed Apps

Reply:
I don't think we should block the spec on the size debate. For me, it's what will be the impact on my system when I uninstall the package.